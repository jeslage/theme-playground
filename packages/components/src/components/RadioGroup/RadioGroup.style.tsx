import styled from 'styled-components';

const StyledRadioGroup = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
`;

export default StyledRadioGroup;
