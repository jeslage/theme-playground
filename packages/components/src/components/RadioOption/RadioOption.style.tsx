import styled from 'styled-components';

const StyledRadioOption = styled.label`
  position: relative;
  cursor: pointer;
  font-size: 12px;

  input {
    position: absolute !important;
    height: 1px;
    width: 1px;
    overflow: hidden;
    visibility: hidden;
    clip: rect(1px 1px 1px 1px); /* IE6, IE7 */
    clip: rect(1px, 1px, 1px, 1px);
    white-space: nowrap; /* added line */
  }

  div {
    display: flex;
    align-items: center;
    padding: 0.5rem 0.75rem;
    margin: 0 0 0 10px;
    border-radius: 3rem;
    opacity: 0.5;
    color: ${({ theme }) => theme.text};
    border: 1px solid ${({ theme }) => theme.border};
    transition: all 0.2s ease-in-out;

    svg {
      width: 20px;
      height: auto;
      fill: ${({ theme }) => theme.text};
    }
  }

  div:hover {
    opacity: 1;
  }

  input:checked + div {
    opacity: 1;
    background: ${({ theme }) => theme.text};
    color: ${({ theme }) => theme.backgroundDark};
  }
`;

export default StyledRadioOption;
