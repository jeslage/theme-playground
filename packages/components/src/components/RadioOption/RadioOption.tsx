import * as React from 'react';

import StyledRadioOption from './RadioOption.style';

export interface RadioOptionProps {
  onChange?: (val: string) => void;
  name?: string;
  value: string;
  label?: string;
  icon?: JSX.Element;
  isChecked?: boolean;
}

const RadioGroup: React.FC<RadioOptionProps> = ({
  label,
  name,
  value,
  onChange,
  isChecked,
  icon
}) => {
  return (
    <StyledRadioOption htmlFor={value}>
      <input
        type="radio"
        id={value}
        name={name}
        value={value}
        checked={isChecked}
        onChange={({ target }) => {
          if (onChange) {
            onChange((target as HTMLInputElement).value);
          }
        }}
      />
      <div>{icon || label || value}</div>
    </StyledRadioOption>
  );
};

export default RadioGroup;
