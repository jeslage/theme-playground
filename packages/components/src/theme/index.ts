export const darkTheme = {
  base: 'dark',
  backgroundDark: '#2f2f2f',
  backgroundLight: '#333',
  border: '#999999',
  text: '#FFFFFF'
};

export const lightTheme = {
  base: 'light',
  backgroundDark: '#f6f9fc',
  backgroundLight: '#fff',
  border: '#666666',
  text: '#333333'
};

export const getUiTheme = theme => {
  if (theme === undefined) return darkTheme;

  if (typeof theme === 'string') {
    if (theme === 'dark') {
      return darkTheme;
    } else if (theme === 'light') {
      return lightTheme;
    }

    throw Error(
      `Theme "${theme}" is not available. Please use "light" or "dark", or provide a custom theme object.`
    );
  }

  if (theme.base && theme.base === 'light') {
    return {
      ...lightTheme,
      ...theme
    };
  }

  return {
    ...darkTheme,
    ...theme
  };
};
