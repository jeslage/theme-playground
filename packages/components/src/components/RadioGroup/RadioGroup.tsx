import * as React from 'react';

import StyledRadioGroup from './RadioGroup.style';
import RadioOption, { RadioOptionProps } from '../RadioOption/RadioOption';
import Label from '../Label/Label';
export interface RadioGroupProps {
  iconBefore?: JSX.Element;
  label: string;
  description?: string;
  onChange: (val: string) => void;
  value: string;
  name: string;
  children?: React.ReactNode;
}

const RadioGroup: React.FC<RadioGroupProps> = ({
  label,
  description,
  iconBefore,
  name,
  value,
  onChange,
  children
}) => {
  return (
    <StyledRadioGroup>
      <Label iconBefore={iconBefore} label={label} description={description} />

      {React.Children.map(
        children,
        (child: React.ReactElement<RadioOptionProps>) => {
          if (child.type !== RadioOption) return;
          return React.cloneElement(child, {
            onChange,
            name,
            label: child.props.label,
            value: child.props.value,
            icon: child.props.icon,
            isChecked: child.props.value === value
          });
        }
      )}
    </StyledRadioGroup>
  );
};

export default React.memo(
  RadioGroup,
  (prev, next) => prev.value === next.value
);
