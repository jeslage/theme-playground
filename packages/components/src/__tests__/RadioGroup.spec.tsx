import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import RadioGroup from '../components/RadioGroup/RadioGroup';
import RadioOption from '../components/RadioOption/RadioOption';
import StyledRadioGroup from '../components/RadioGroup/RadioGroup.style';
import { darkTheme } from '../theme';

describe('RadioGroup', () => {
  it('RadioGroup should render correctly', () => {
    const component = shallow(
      <RadioGroup
        label="Active Theme"
        name="themes"
        value={name}
        onChange={val => console.log(val)}
      >
        <RadioOption label="default" value="default" />
        <RadioOption label="primary" value="primary" />
      </RadioGroup>
    );

    expect(component).toMatchSnapshot();
  });

  it('RadioGroup should render nothing', () => {
    const component = shallow(
      <RadioGroup
        label="Active Theme"
        name="themes"
        value={name}
        onChange={val => console.log(val)}
      >
        <p>Another child</p>
      </RadioGroup>
    );

    expect(component).toMatchObject({});
    expect(component).toMatchSnapshot();
  });

  it('StyledRadioGroup should render correctly', () => {
    const component = shallow(
      <StyledRadioGroup theme={{ colors: darkTheme }} />
    );

    expect(component).toMatchSnapshot();
  });
});
