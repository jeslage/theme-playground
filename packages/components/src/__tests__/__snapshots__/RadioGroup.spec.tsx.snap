// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup RadioGroup should render correctly 1`] = `
<styled.div>
  <Label
    label="Active Theme"
  />
  <RadioGroup
    isChecked={false}
    key=".0"
    label="default"
    name="themes"
    onChange={[Function]}
    value="default"
  />
  <RadioGroup
    isChecked={false}
    key=".1"
    label="primary"
    name="themes"
    onChange={[Function]}
    value="primary"
  />
</styled.div>
`;

exports[`RadioGroup RadioGroup should render nothing 1`] = `
<styled.div>
  <Label
    label="Active Theme"
  />
</styled.div>
`;

exports[`RadioGroup StyledRadioGroup should render correctly 1`] = `
<div
  className="sc-AxjAm iLVQmu"
/>
`;
